src/mono/wasm/runtime/library_mono.js

```diff
@@ -484,8 +512,10 @@ var MonoSupportLib = {
 			}
 		},
 
-		mono_text_decoder: undefined,
 		string_decoder: {
+			result: undefined,
```

Comment:
I'm not clear on the value of explicitly defining properties as undefined, is this fixing something?

Reply:
Strict mode needs variables to be declared before being assigned. For example `t = 5` throws and error whereas `var t = 5` doesn't. However, to prevent accidentally breaking something by giving variables premature values, I assign them to be undefined. This is still accepted by strict mode.

Reply:
These are properties not variables though?

Reply:
Yes, I just tried it without these and it looks like it works. I'll push a fix. Thanks

Reply:
Open to going back and doing this in a follow-up PR, just don't want to bundle it in with this if it isn't necessary